libraries/matrix/test/src/main/kotlin/io/element/android/libraries/matrix/test/FakeMatrixClient.kt

```diff
@@ -77,6 +77,8 @@ class FakeMatrixClient(
     private val clearCacheLambda: () -> Unit = { lambdaError() },
     private val userIdServerNameLambda: () -> String = { lambdaError() },
     private val getUrlLambda: (String) -> Result<String> = { lambdaError() },
+    var isNativeSlidingSyncSupportedLambda: suspend () -> Boolean = { true },
+    var isUsingNativeSlidingSyncLambda: () -> Boolean = { true },
```

Comment:
I prefer to use `private val` for lambda like this. Also, they seem to be unused, maybe add a quick unit test to check the value of `needsSlidingSyncMigration`?

Reply:
I usually prefer using the `public var` because that way you can set up both the value on the constructor or later in the tests if you need to change it without having to create a new method just for setting it.